mesa/mesaPROTON_OC.py

```diff
@@ -426,6 +432,147 @@ def setup_siblings(self):
             for x in targets:
                 x.addSiblingLinks(p)
 
+    def generate_households(self):
+        self.families = list()
+        head_age_dist = self.read_csv_city("head_age_dist_by_household_size")
```

Comment:
The main reason we turn into a list is because of the function: extra.pick_from_pair_list().  This function often operates in distributions that are loaded by csv and takes as argument a list. 
Possible solutions:
1. Create a function that works directly with pandas to replace extra.pick_from_pair_list()
2. Manage csv files with another library (e.g. numpy)

Reply:
I replaced pandas with a nested dictionary. Now the procedure is much faster. 
Testing with 10,000 agents:
Pandas = 52 minutes
<a href="https://imgbb.com/"><img src="https://i.ibb.co/b5wFJj9/Annotazione-2020-07-15-171703.png" alt="Annotazione-2020-07-15-171703" border="0"></a>
Nested dictionary = 4 minutes
<a href="https://imgbb.com/"><img src="https://i.ibb.co/vmWyGq0/Annotazione-2020-07-16-114043.png" alt="Annotazione-2020-07-16-114043" border="0"></a>